src/connector/src/sink/iceberg/compaction.rs

```diff
@@ -0,0 +1,172 @@
+    pub fn new(config: NimtableCompactionConfig) -> Self {
+        let sdk_config = aws_types::SdkConfig::builder()
+            .behavior_version(BehaviorVersion::v2024_03_28())
+            .region(Some(Region::new(config.region.clone())))
+            .credentials_provider(SharedCredentialsProvider::new(
+                aws_credential_types::Credentials::from_keys(
+                    config.access_key.clone(),
+                    config.secret_key.clone(),
+                    None,
+                ),
+            ))
+            .build();
+        Self {
+            client: Client::new(&sdk_config),
+            config,
+        }
+    }
+
+    async fn wait_job_finish(&self, job_run_id: String) -> anyhow::Result<()> {
+        let start_time = Instant::now();
+        let success_job_run = loop {
+            let output = self
+                .client
+                .get_job_run()
+                .job_run_id(&job_run_id)
+                .application_id(self.config.application_id.clone())
+                .send()
+                .await?;
+            let job_run = output.job_run.ok_or_else(|| anyhow!("empty job run"))?;
+            match &job_run.state {
+                JobRunState::Cancelled | JobRunState::Cancelling | JobRunState::Failed => {
+                    return Err(anyhow!(
+                        "fail state: {}. Detailed: {}",
+                        job_run.state,
+                        job_run.state_details
+                    ));
+                }
+                JobRunState::Pending
+                | JobRunState::Queued
+                | JobRunState::Running
+                | JobRunState::Scheduled
+                | JobRunState::Submitted => {
+                    info!(
+                        elapsed = ?start_time.elapsed(),
+                        job_status = ?job_run.state,
+                        "waiting job."
+                    );
+                    sleep(Duration::from_secs(5)).await;
+                }
+                JobRunState::Success => {
+                    break job_run;
+                }
+                state => {
+                    return Err(anyhow!("unhandled state: {:?}", state));
+                }
+            };
+        };
+        info!(
+            job_run_id,
+            details = success_job_run.state_details,
+            elapsed = ?start_time.elapsed(),
+            "job run finish"
+        );
+        Ok(())
+    }
+
+    pub async fn compact(
+        &self,
+        warehouse: String,
+        db: String,
+        table: String,
+    ) -> anyhow::Result<()> {
+        let start_result = self
+            .client
+            .start_job_run()
+            .name(format!(
+                "job-run-{}",
+                SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_millis()
+            ))
+            .application_id(self.config.application_id.clone())
```

Comment:
The documentation says that [client_token](https://docs.rs/aws-sdk-emrserverless/latest/aws_sdk_emrserverless/struct.Client.html#method.start_job_run) is required but we didn't set it here.

Reply:
From the document, it seems that the client token is used to ensure idempotent, which means if two requests are sent with a same client token, the run will be run for only once. 

In my local testing, the test can run successfully without this client token, so I think the server will disable this idempotent check when it's absent, and the run can be started anyway.

Reply:
The aws-sdk-emrserverless documentation is inaccurate then lol